src/routes/auth.route.ts

```diff
@@ -1,33 +1,58 @@
 import * as express from "express";
 import * as passport from "passport";
-// import { AppError } from "../exceptions/ApplicationError";
-import userValidation from "../validation/user.validation";
+import ValidationError from "../exceptions/ValidationError";
+import * as userValidation from "../validation/user.validation";
 import AuthService from "../services/auth.service";
 import BaseRoute from "../routes/base.router";
-import User from "../db/models/User";
+import { ISigninUser, IUser } from "../db/models/User";
 
-const authenticateLocal = passport.authenticate("local", { session: false });
-const authenticateToken = passport.authenticate("jwt", { session: false });
+// const authenticateLocal = passport.authenticate("local", { session: false });
+// const authenticateToken = passport.authenticate("jwt", { session: false });
 
-const router = express.Router();
-
-export default class AuthRouter extends BaseRoute {
+class AuthRouter extends BaseRoute {
 
 	authService: AuthService;
 
 	constructor() {
 		super();
 		this.authService = new AuthService();
+		this.initializeRoutes();
 	}
 	async signinRoute(req: express.Request, res: express.Response, next: express.NextFunction) {
-		const { user } = req;
+		const user: ISigninUser = req.body;
 		try {
-			if (user) {
-				const result = await this.authService.signin(user);
-			}
+			await userValidation.default.signin.validateAsync(user, { abortEarly: false });
+		} catch (err) {
+			return next(new ValidationError(err.details))
+		}
+		try {
+			const result: any = await this.authService.signin(user);
 			return res.status(result.statusCode || 200).send(result);
 		} catch (err) {
 			return next(err)
 		}
 	}
-}
\ No newline at end of file
+
+	async signupRoute(req: express.Request, res: express.Response, next: express.NextFunction) {
+		const user: IUser = req.body;
+		try {
+			await userValidation.default.signup.validateAsync(user, { abortEarly: false });
+		} catch (err) {
+			return next(new ValidationError(err.details))
+		}
+		try {
+			const result: any = await this.authService.signup(user);
```

Comment:
Probably you would like to have explict type here instead `any`
http://knexjs.org/#Builder-insert
> // Returns [1] in "mysql", "sqlite", "oracle"; [] in "postgresql" unless the 'returning' parameter is set.

```suggestion
			const result: [] = await this.authService.signup(user);
```